### PR TITLE
dedupe Alt-Svc header values

### PR DESCRIPTION
**Pull request by mattrobenolt**
Addresses https://github.com/lucas-clemente/quic-go/pull/3432#issuecomment-1164796769

### CODE REVIEW

File: http3/server.go [hunk 1]
```diff
@@ -313,6 +313,15 @@ func extractPort(addr string) (int, error) {
 	return portInt, nil
 }
 
+func findString(haystack []string, needle string) int {
```
Comment (marten-seemann) [inline http3/server.go:316]: You're not using the `int` return value. What about `contains([]string, string) bool` instead?

Comment (marten-seemann) [reply http3/server.go:316]: Or just inline the entire function, then we don't need to think about naming ;)

**Review by marten-seemann — COMMENTED** [review]
Very nice find! Thank you for the PR, just a minor stylistic comment.

**Comment by marten-seemann** [comment]
Could you also add a test case for this please?

**Comment by marten-seemann** [comment]
@mattrobenolt We're planning to cut a new quic-go release really soon. Would be great if we could get your patch in. Will you be able to incorporate the comments?

**Comment by mattrobenolt** [comment]
@marten-seemann test added, should be good to go.

**Review by marten-seemann — APPROVED** [review]
Thanks!

**Pull request merged**